### ISSUE TITLE
df.convert_objects removed from pandas

### ISSUE THREAD

**Issue by philipstarkey** [labscript-suite/lyse#52]
**Original report ([archived issue](https://labscript-suite.github.io/labscript-suite-bitbucket-archive/#!/labscript_suite/lyse/issues/52)) by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
pandas 0.25 has dropped `DataFrame.convert_obects()`, resulting in an exception from the server when getting the dataframe using `lyse.data()`.

```python
AttributeError: 'DataFrame' object has no attribute 'convert objects'
```

Discussion about the deprecation and removal here:

https://github.com/pandas-dev/pandas/issues/11221

As a reminder, we're using this function to convert columns of the dataframe from Python objects into numpy/pandas dtypes where possible, which makes the dataframe faster to pickle and send over the wire.

We'll need to decide on what to do. It is possible that the performance reason for doing `convert_objects()` may no longer be as important as performance in other relevant components may have improved, though it is still a semantic change to be returning dataframes where the numpy arrays pulled out of them are of dtype `object` containing Python floats instead of being dtype `float` as expected.

It seems like the alternatives to `convert_objects` may require explicitly saying the type of each column, which would be super annoying. But I'll look into it and see if we can replicate the current behaviour using the alternatives.

**Comment by zakv** [comment]
I believe this issue has been resolved and can be closed

**Comment by zakv** [comment]
Also, this is more or less a duplicate of #34 so I think both can be closed.

**Comment by philipstarkey** [comment]
Yep, resolved in 9fac3f535 ([BitBucket PR 70](https://bitbucket-archive.labscriptsuite.org/#!/labscript_suite/lyse/pull-requests/70/page/1))
